semantic/lookup.go

```diff
@@ -30,70 +30,33 @@ func LookupBuiltInType(pkg, name string) (MonoType, error) {
 	return monotype, nil
 }
 
-// lookup is a helper function that performs a lookup of a package identifier in
-// a flatbuffers TypeEnvironment. It first checks for the package using path string
-// and then checks for the package identifier, the "name" string in this case, returning
-// the corresponding MonoType if found
-func lookup(env *fbsemantic.TypeEnvironment, pkg, name string) (*fbsemantic.Prop, error) {
-	// Find package
-	typeAssign := new(fbsemantic.TypeAssignment)
-	if pkgErr := foundPackage(env, typeAssign, pkg); pkgErr != nil {
-		return nil, pkgErr
-	}
-
-	// Grab PolyType expr; expr should always be of type MonoTypeRow
-	polytype := typeAssign.Ty(nil)
-	if polytype.ExprType() != fbsemantic.MonoTypeRow {
-		return nil, errors.Newf(
-			codes.Internal,
-			"Expected PolyType expr to be fbsemantic.MonoTypeRow; found %v",
-			polytype.ExprType(),
-		)
-	}
+func TypeEnvMap(env *fbsemantic.TypeEnvironment) map[string]map[string]*fbsemantic.Prop {
+	envMap := make(map[string]map[string]*fbsemantic.Prop)
 	var table flatbuffers.Table
-	if !polytype.Expr(&table) {
-		return nil, errors.New(codes.Internal, "PolyType expr is not valid")
-	}
-
-	// check for package identifier in Row Props
-	row := new(fbsemantic.Row)
-	row.Init(table.Bytes, table.Pos)
-	prop := new(fbsemantic.Prop)
-	if propErr := foundProp(row, prop, name); propErr != nil {
-		return nil, propErr
-	}
-
-	return prop, nil
-}
-
-// foundPackage is a helper function that iterates over type assignments and checks
-// for a given package returning an error if the package is not found
-func foundPackage(env *fbsemantic.TypeEnvironment, obj *fbsemantic.TypeAssignment, pkg string) error {
 	l := env.AssignmentsLength()
+
 	for i := 0; i < l; i++ {
-		if !env.Assignments(obj, i) {
-			return errors.Newf(codes.Internal, "package %v not found; last position %v", pkg, i)
-		} else {
-			if string(obj.Id()) == pkg {
-				return nil
-			}
+		newAssign := new(fbsemantic.TypeAssignment)
+		_ = env.Assignments(newAssign, i) // this call assigns a value to newAssign
+		assignId := string(newAssign.Id())
+		polytype := newAssign.Ty(nil)
+		if !polytype.Expr(&table) {
+			panic(fmt.Errorf("PolyType does not have a MonoType; something went wrong %v", string(polytype.ExprType())))
 		}
```

Comment:
I wanted to keep this panic as descriptive/helpful as possible. I don't really expect that this is a case that we would hit.

Reply:
Makes sense, I agree this is very unlikely to happen.  I think that `string(polytype.ExprType())` will just produce a stringified small integer, like `"1"` or something though.  (FYI, there is a map called `EnumNamesMonoType` that maps it to a readable name). You might just put `newAssign.Id()` in there instead, that would indicate which symbol had the problem.

Reply:
Yeah, the map is a good idea. I'll add the `assignId` and the `ExprType` to the panic statement.